File: source-connector/src/test/java/sourceconnector/repository/file/filter/FileExtensionFilterTest.java
package sourceconnector.repository.file.filter;

import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import sourceconnector.domain.file.LocalFileKey;

import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.Collections;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

class FileExtensionFilterTest {

  @DisplayName("Should accept files with specified extensions")
  @Test
  void accept() {
    // given
    FileFilter filter = new FileExtensionFilter(List.of(".ndjson", ".md"));
    // when
    var result1 = filter.accept(LocalFileKey.from(Path.of("test.ndjson")));
    var result2 = filter.accept(LocalFileKey.from(Path.of("README.md")));
    var result3 = filter.accept(LocalFileKey.from(Path.of("image.png")));
    var result4 = filter.accept(LocalFileKey.from(Path.of("temp.tmp")));

    // then
    assertTrue(result1);
    assertTrue(result2);
    assertFalse(result3);
    assertFalse(result4);
  }

  @DisplayName("Should not be empty extensions")
  @Test
  void emptyExtensionTest() {
    Assertions.assertThatThrownBy(() -> new FileExtensionFilter(Collections.emptyList()))
      .isInstanceOf(IllegalArgumentException.class)
      .hasMessage("file extensions cannot be null or empty");
  }
}

File: source-connector/src/test/kotlin/sourceconnector/repository/file/filter/FileExtensionFilterTest.kt
package sourceconnector.repository.file.filter

import offsetmanager.domain.file.LocalFileKey.Companion.from
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import java.nio.file.Path

internal class FileExtensionFilterTest {
  @DisplayName("Should accept files with specified extensions")
  @Test
  fun accept() {
    // given
    val filter: FileFilter = FileExtensionFilter(listOf(".ndjson", ".md"))
    // when
    val result1 = filter.accept(from(Path.of("test.ndjson")))
    val result2 = filter.accept(from(Path.of("README.md")))
    val result3 = filter.accept(from(Path.of("image.png")))
    val result4 = filter.accept(from(Path.of("temp.tmp")))

    // then
    Assertions.assertTrue(result1)
    Assertions.assertTrue(result2)
    Assertions.assertFalse(result3)
    Assertions.assertFalse(result4)
  }

  @DisplayName("Should not be empty extensions")
  @Test
  fun emptyExtensionTest() {
    org.assertj.core.api.Assertions.assertThatThrownBy {
      FileExtensionFilter(
        emptyList()
      )
    }
      .isInstanceOf(IllegalArgumentException::class.java)
      .hasMessage("file extensions cannot be null or empty")
  }
}
